### PR TITLE
Make integration tests using test suites

### PR DESCRIPTION
**Pull request by xiangfu0**
Split integration tests into two suites: `integration-suite-1` and `integration-suite-2`

### CODE REVIEW

**Review by walterddr — COMMENTED** [review]
are we planning to reuse cluster for some of the SQL tests or just to use this framework to better balance the 2 integration test sides in GHA?

**Comment by xiangfu0** [comment]
> are we planning to reuse cluster for some of the SQL tests or just to use this framework to better balance the 2 integration test sides in GHA?

Integration 1 and 2 are for balancing.
The custom tests is a suite to reuse the test.

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/FloatingPointDataTypeTest.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
 /**
  * Integration test for floating point data type (float & double) filter queries.
  */
-@Test(suiteName = "CustomClusterIntegrationTest")
+@Test(suiteName = "CustomClusterIntegrationTest", groups = {"custom-integration-suite"})
```
Comment (walterddr) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/FloatingPointDataTypeTest.java:42]: so before this change the @BeforeSuite and @BeforeClass is basically the same b/c they are not group into suite is that correct? (we should see huge test time reduction if so)

Comment (xiangfu0) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/custom/FloatingPointDataTypeTest.java:42]: Only this package has the suite implementation, which is already tested using `mvn test -P github-actions,custom-cluster-integration-test-suite || exit 1`. But in this case, we are missing the code coverage report for this.
So just wanna merge these two suite to run together.

**Comment by xiangfu0** [comment]
We have 3 integration tests profile right now:
1. integration-tests-1
2. integration-tests-2
3. custom-integration-tests

1 and 2 are just for rebalancing
3 is a suite to bring up cluster once

Right now we split the integration tests in two parts:
For 1, we run two commands:
```
mvn test -P github-actions,custom-cluster-integration-test-suite
mvn test -P github-actions,integration-test-suite-1
```
For 2, just one command:
```
mvn test -P github-actions,integration-test-suite-2
```

The issue with 1 is that we are missing the code coverage report for the first command.
Hence we need to model everything into test suites then we can combine and run two suites together.

File: pinot-integration-tests/src/test/integration-test-suite-1.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="integration-suite-1">
+  <test name="integration-suite-1">
+    <groups>
+      <run>
+        <include name="integration-suite-1"/>
+      </run>
+    </groups>
+    <packages>
+      <package name="org.apache.pinot.integration.tests.*"/>
```
Comment (walterddr) [inline pinot-integration-tests/src/test/integration-test-suite-1.xml:30]: this looks like in includes everything (*) and the other `*-2` also have the same package name, do we plan to still do the first letter split?

Comment (walterddr) [reply pinot-integration-tests/src/test/integration-test-suite-1.xml:30]: nvm i actually see you do exclusion from the `*-2` xml... (maybe to call it out in comment)

File: pinot-integration-tests/src/test/integration-test-suite-2.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    under the License.
+
+-->
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="integration-suite-2">
+  <test name="integration-suite-2">
+    <groups>
+      <run>
+        <exclude name="custom-integration-suite"/>
+        <exclude name="integration-suite-1"/>
```
Comment (walterddr) [inline pinot-integration-tests/src/test/integration-test-suite-2.xml:27]: good to call out that integration-suite-2 is a catch all and theoretically it doesn't need name annotation in `@Test`

**Comment by walterddr** [comment]
lgtm can you fix the test failures

**Pull request closed without merging**